### PR TITLE
adding a function that prints the binary of a number

### PR DESCRIPTION
**Pull request by Mariamahmed44**

### CODE REVIEW

**Review by abadr99 — CHANGES_REQUESTED** [review]
Hi @Mariamahmed44 ,
Looks good so far. But I think it will be useful if you add some test cases to make sure if it works good or not, also to illustrate how can we use this function. 
Testing special cases like 0 and 255 will be good :-)

**Comment by Mariamahmed44** [comment]
@abadr99 Done

File: C/regression-tests/regression-tests/gpio/main.c [hunk 1]
```diff
@@ -123,4 +123,14 @@ int main()
     print_str("DDRB_REG: ");
     print_num(DDRB_REG);
     print_str("\n");
+    print_num_binary(0);
+    print_str("\n");
+    print_num_binary(255);
+    print_str("\n");
+    print_num_binary(16);
```
Comment (abadr99) [inline C/regression-tests/regression-tests/gpio/main.c:130]: I think it will be good if the number is prefixed by `0b` as indication for binary numbers

**Review by abadr99 — CHANGES_REQUESTED** [review]

**Comment by Mariamahmed44** [comment]
++done;

**Review by abadr99 — APPROVED** [review]

**Pull request merged**